concordium-contracts-common/src/types.rs

```diff
@@ -627,6 +661,18 @@ impl fmt::Display for Duration {
 #[cfg_attr(feature = "fuzz", derive(Arbitrary))]
 pub struct AccountAddress(pub [u8; ACCOUNT_ADDRESS_SIZE]);
 
+#[cfg(feature = "concordium-quickcheck")]
+impl quickcheck::Arbitrary for AccountAddress {
+    fn arbitrary(g: &mut Gen) -> AccountAddress {
+        //todo should it satisfy any properties, or are random bytes okay?
```

Comment:
TODO.

To my knowledge, random bytes should be ok. We don't check the version or checksum in contracts.
On-chain, the first byte is the version (always `1` currently), the next 27 bytes are the payload, and the last four bytes are the first four bytes of `sha256(sha256(version + payload))`. More info here: https://en.bitcoin.it/wiki/Base58Check_encoding (bitcoin uses `0` for the version).

Reply:
The account address is by definition a SHA256(credId) where credId is the credential registration ID of the first credential (it is 48 bytes, and an encoding of some group element).

Hence an account address has no structure.

The String representation of the account address uses base58check encoding, and that one is defined by prepending a 1 byte, and hashing the result twice, then appending the last 4 bytes of the hash. Then encoding that in base58.

But an account address itself has no structure.

Reply:
So, it's ok to keep random bytes, right?

Reply:
Yes.

Reply:
Done. I left it as it is and removed the comment.